runtime/compiler/env/ClassLoaderTable.cpp

```diff
@@ -225,26 +225,18 @@ TR_PersistentClassLoaderTable::associateClassLoaderWithClass(J9VMThread *vmThrea
                nameLength, (const char *)name, loader
             );
 #endif /* defined(J9VM_OPT_JITSERVER) */
-         // TODO: for now we bail out right here, but when we actually need to track names
-         // without the chains being present this return must be removed.
-         return;
          }
       else
          {
          chain = _sharedCache->pointerFromOffsetInSharedCache(chainOffset);
          TR_ASSERT(_sharedCache->isPointerInSharedCache(chain), "Class chain must be in SCC");
          }
       }
-   else
-      {
-      // TODO: for now we bail out right here, but when we actually need to track names
-      // without an SCC being present at all this return must be removed.
-      return;
-      }
 
-   // If we are using the JITServer AOT cache, we need to remember the name of clazz with or without chain.
-   // If we aren't using it, there is no point in continuing.
-   if (!chain && !useAOTCache)
+   // If we are using the JITServer AOT cache and ignoring the local SCC, we need to remember the name of clazz
+   // with or without chain. Otherwise (not using AOT cache or not ignoring the local SCC) there is no point in continuing
+   // without a chain.
+   if (!chain && (!useAOTCache || !_persistentMemory->getPersistentInfo()->getJITServerAOTCacheIgnoreLocalSCC()))
```

Comment:
This doesn't compile without `defined(J9VM_OPT_JITSERVER)`, e.g. on macOS; see [failed build](https://openj9-jenkins.osuosl.org/job/Build_JDK22_aarch64_mac_Personal/9):
```
[2024-02-20T21:51:46.296Z] /Users/jenkins/workspace/Build_JDK22_aarch64_mac_Personal/openj9/runtime/compiler/env/ClassLoaderTable.cpp:239:76: error: no member named 'getJITServerAOTCacheIgnoreLocalSCC' in 'TR::PersistentInfo'
[2024-02-20T21:51:46.296Z]    if (!chain && (!useAOTCache || !_persistentMemory->getPersistentInfo()->getJITServerAOTCacheIgnoreLocalSCC()))
[2024-02-20T21:51:46.296Z]                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ^
[2024-02-20T21:51:46.296Z] 1 error generated.
[2024-02-20T21:51:46.296Z] make[6]: *** [runtime/compiler/CMakeFiles/j9jit.dir/env/ClassLoaderTable.cpp.o] Error 1
```

Reply:
I'll put a fix shortly